pkg/ingester/ingester_v2.go

```diff
@@ -1114,14 +1131,23 @@ func (i *Ingester) openExistingTSDB(ctx context.Context) error {
 		return filepath.SkipDir // Don't descend into directories
 	})
 
+	if walkErr != nil {
```

Comment:
This cannot happen currently, since walkFn will filter out any error. (see other comment)

Reply:
Given I addressed the other comment, why can't happen? The `Walk()` interrupts and returns error as soon as we return error. Errors returned by `Walk()` itself are not filtered again via `Walk()`.

Reply:
Now it can.

Reply:
Even before it could, in case `os.Open(path)` or `f.Readdirnames(1)` failed.

Reply:
You are right.